luigi/task.py

```diff
@@ -284,6 +284,10 @@ def task_module(self):
         # TODO(erikbern): we should think about a language-agnostic mechanism
         return self.__class__.__module__
 
+    @property
+    def run_on_main_process(self):
```

Comment:
Please add docs here. And also warn users that this mode should be avoided whenever possible, because any blocking IO will make the keep-alive-thread not run.